backend-rust/src/graphql_api/account_metrics.sql

```diff
@@ -1,42 +1,54 @@
+-- accounts (i.e. the account index) at or before (i.e. <) the start of the
 -- bucket and the same number just before (i.e. <) the next bucket. The
 -- difference between the two numbers should give the total number of accounts
 -- created within the bucket.
-SELECT
-    -- The bucket time is the starting time of the bucket.
-    bucket_time as "bucket_time!",
-    -- Number of accounts at or before the bucket.
-    COALESCE(before_bucket.index, 0) as start_index,
-    -- Number of accounts at the end of the bucket.
-    COALESCE(after_bucket.index, 0) as end_index
-FROM
-    -- We generate a time series of all the buckets where accounts will be counted.
-    -- $1 is the full period, $2 is the bucket interval.
-    -- For the rest of the comments, let's go with the example of a full period of 7 days with 6 hour buckets.
-    generate_series(
-        -- The first bucket starts 7 days ago.
-        now() - $1::interval,
-        -- The final bucket starts 6 hours ago, since the bucket time is the start of the bucket.
-        now() - $2::interval,
-        -- Each bucket is seperated by 6 hours.
-        $2::interval
+WITH
+  buckets AS (
+    SELECT generate_series(
+      now() - $1::interval,
+      now() - $2::interval,
+      $2::interval
     ) AS bucket_time
-LEFT JOIN LATERAL (
-    SELECT accounts.index
-    FROM accounts
-    LEFT JOIN transactions on transaction_index = transactions.index
-    LEFT JOIN blocks ON transactions.block_height = height
-    WHERE slot_time < bucket_time
-    ORDER BY slot_time DESC
-    LIMIT 1
-) before_bucket ON true
-LEFT JOIN LATERAL (
-    SELECT accounts.index
-    FROM accounts
-    LEFT JOIN transactions on transaction_index = transactions.index
-    LEFT JOIN blocks ON transactions.block_height = height
-    WHERE slot_time < bucket_time + $2::interval
-    ORDER BY slot_time DESC
-    LIMIT 1
-) after_bucket ON true
+  ),
+  thresholds AS (
+    SELECT
+      b.bucket_time,
+      -- Find the latest transaction index before the start of the bucket
+      (
+        SELECT t.index
+        FROM transactions t
+        JOIN blocks     bl ON bl.height = t.block_height
+        WHERE bl.slot_time < b.bucket_time
+        ORDER BY bl.slot_time DESC
+        LIMIT 1
+      ) AS tx_start,
+      -- Find the latest transaction index before the end of the bucket
+      (
+        SELECT t.index
+        FROM transactions t
+        JOIN blocks     bl ON bl.height = t.block_height
+        WHERE bl.slot_time < b.bucket_time + $2::interval
+        ORDER BY bl.slot_time DESC
+        LIMIT 1
+      ) AS tx_end
+    FROM buckets b
+  )
+SELECT
+  th.bucket_time AS "bucket_time!",
+  COALESCE(
+    (
+      SELECT MAX(a.index)
+      FROM accounts a
+      WHERE a.transaction_index <= th.tx_start
+    ), 0
+  ) AS "start_index!",
+  COALESCE(
+    (
+      SELECT MAX(a.index)
+      FROM accounts a
+      WHERE a.transaction_index <= th.tx_end
```

Comment:
I do not entirely understand why this is fast enough when not having an index on transaction_index. It does seem to perform much better than dotnet version even without the index though. So lets skip it.

Reply:
I can leave a comment that this is a quick win for performance gains.

Reply:
I am refraining from doing it because from a user perspective performance is more than fine, so it only increases the amount of indexing which has to occur.

As long as this is fine from a user perspective it is fine and good, and we should not waste resources on maintaining something which is not needed. 

So my point being:
If you want to create such an index, in case the amount of accounts suddenly increases much, then I will just create the index instead of creating the task. 

As it sounds like you want that I will do that, as I kind of get the argument both ways.

Reply:
Not doing it now, means that this might come back to us as technical dept, delaying other projects.
Since it is so low effort, and you already have the mental model build, I would prefer to go for it already.